backend/lucene/src/main/java/org/hibernate/search/backend/lucene/lowlevel/docvalues/impl/DoubleMultiValuesToSingleValuesSource.java

```diff
@@ -176,21 +224,85 @@ public boolean advanceExact(int parentDoc) throws IOException {
 				}
 
 				lastSeenParentDoc = parentDoc;
-				lastEmittedValue = mode.pick( values, childDocs, nextChildWithValue, parentDoc );
-				return true;
+
+				if ( mode == null || mode == MultiValueMode.NONE ) {
+					all = list( values, childDocs, nextChildWithValue, parentDoc );
+				}
+				else {
+					OptionalDouble pick = mode.pick( values, childDocs, nextChildWithValue, parentDoc );
+					if ( pick.isPresent() ) {
+						all = Collections.singletonList( pick.getAsDouble() );
+					}
+				}
+
+				if ( all != null && !all.isEmpty() ) {
+					value = all.iterator();
+					return true;
+				}
+				return false;
 			}
 
 		};
 	}
 
+	protected List<Double> list(SortedNumericDoubleDocValues values, DocIdSetIterator docItr, int startDoc, int endDoc) throws IOException {
+		List<Double> result = new ArrayList<>();
+		int count = 0;
+		for ( int doc = startDoc; doc < endDoc; doc = docItr.nextDoc() ) {
+			if ( values.advanceExact( doc ) ) {
+				int docValueCount = values.docValueCount();
+				for ( int i = 0; i < docValueCount; i++ ) {
+					result.add( values.nextValue() );
+				}
+			}
+		}
+
+		Collections.sort( result );
+		return result;
+	}
+
+	protected NumericDoubleValues replaceScores(NumericDoubleValues values, DoubleValues scores) {
```

Comment:
This is not necessary and it's actually the source of your troubles related to missing values on sorts. You just don't need this hack, and without this hack you don't need your commits "Fix sort missing values".

I will remove this before submitting a new PR.